### PR TITLE
[BLOG] doctesting for pydata libraries

### PR DESCRIPTION
**Pull request by Sheila-nk**
## Text styling

- [x] The blog is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [x] If there are headers, they use the proper header tags in order to do so (with only one level-one header).
- [x] All links describe where they link to (for example, check the [Quansight labs website](https://labs.quansight.org/)).
- [x] Any kind of styling that the author uses (for example, bold for emphasis) is consistent throughout the blog.

## Non-text contents

- [x] Blog post featured image is in PNG or JPEG format, **not** SVG.
- [x] All content is represented as text (for example, images need alt text and videos need captions or descriptive transcripts).
- [x] If there are emojis, there are not more than three in a row.
- [x] Don't use [flashing gifs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [x] If it were to be read as plain text, the blog still makes sense and no information is missing.

### CODE REVIEW

File: apps/labs/posts/doctesting-for-pydata-libraries.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
+---
+title: 'Doctesting for PyData Libraries'
+author: [Sheila-nk]
```
Comment (pavithraes) [inline apps/labs/posts/doctesting-for-pydata-libraries.md:3]:
```suggestion
authors: [sheila-kahwai]
```

Added you to Storyblok, this is the generated slug. :)

File: apps/labs/posts/doctesting-for-pydata-libraries.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+published: September 30, 2023
+description: 'The journey of a PyData Newbie'
+category: [Doctests]
```
Comment (pavithraes) [inline apps/labs/posts/doctesting-for-pydata-libraries.md:6]:
```suggestion
category: [Developer workflows]
```

We haven't defined a `Doctests` category for the blog, so suggesting `Developer workflows` - which I think is a good fit here.

File: apps/labs/posts/doctesting-for-pydata-libraries.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+With the bugs and vulnerabilities exposed during this process, I was able to refine the plugin further. I then created a [pull request](https://github.com/scipy/scipy/pull/19242) to demonstrate how the pytest plugin could be seamlessly integrated into SciPy. The process is fairly straightforward:
+
+1. **Installation**: Install the plugin via pip.
+2. **Configuration**: Customize your doctesting through a `conftest.py` file.
+3. **Running Doctests in SciPy**: If you're running doctests on SciPy, execute the command `python dev.py test --doctests` in your shell.
+4. **Running Doctests on Other Packages**: If you're not working with SciPy, use the command `pytest --pyargs <your-package> --doctest-modules` to run your doctests.
+
+Voila! :tada:
```
Comment (pavithraes) [inline apps/labs/posts/doctesting-for-pydata-libraries.md:72]:
```suggestion
Voila! 🎉
```

File: apps/labs/posts/doctesting-for-pydata-libraries.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+Once all the technical juggling was done, it was time for what my mentor called "[dogfooding](https://www.forbes.com/sites/michaeldefranco/2014/03/04/not-eating-your-own-dog-food-you-probably-should-be-2/?sh=69c002d0692e)"(a term he picked up from Joel Spolsky's [essay](https://www.joelonsoftware.com/2001/05/05/what-is-the-work-of-dogs-in-this-country/)). The term simply means putting your own product to the test by using it, and I had to make sure that the plugin functioned as expected. I did this by locally running doctests on SciPy's modules. It was an eye-opener, exposing issues like faulty collection – for example, the plugin wasn't collecting compiled and NumPy universal functions for doctesting.
```
Comment (pavithraes) [inline apps/labs/posts/doctesting-for-pydata-libraries.md:63]:
```suggestion
Once all the technical juggling was done, it was time for what my mentor called "[dogfooding](https://www.forbes.com/sites/michaeldefranco/2014/03/04/not-eating-your-own-dog-food-you-probably-should-be-2/?sh=69c002d0692e)" (a term he picked up from Joel Spolsky's [essay](https://www.joelonsoftware.com/2001/05/05/what-is-the-work-of-dogs-in-this-country/)). The term simply means putting your own product to the test by using it, and I had to make sure that the plugin functioned as expected. I did this by locally running doctests on SciPy's modules. It was an eye-opener, exposing issues like faulty collection – for example, the plugin wasn't collecting compiled and NumPy universal functions for doctesting.
```

File: apps/labs/posts/doctesting-for-pydata-libraries.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+Voila! :tada: 
+
+<p align="center">
+    <img
+     alt="An image featuring Kamala Harris on a phone call, with the phrase 'We did it, Joe' displayed at the bottom of the image. Adjacent to the image are the pytest logo, a plus sign, and the text representing the doctesting package 'SCPDT'."
+     src="/posts/doctesting-for-pydata-libraries/we-did-it.png" />
+</p>
+
+## Future Goals
+I am currently in the process of integrating the plugin into SciPy; for more details, you can check out the PR linked [here](https://github.com/scipy/scipy/pull/19242). Looking ahead, our goal is to publish the plugin on [PyPI](https://pypi.org/) and extend its integration to NumPy and other PyData libraries.
```
Comment (pavithraes) [inline apps/labs/posts/doctesting-for-pydata-libraries.md:81]:
```suggestion
I am currently in the process of integrating the plugin into SciPy; for more details, you can check out [the PR](https://github.com/scipy/scipy/pull/19242). Looking ahead, our goal is to publish the plugin on [PyPI](https://pypi.org/) and extend its integration to NumPy and other PyData libraries. 
```

For better accessibility, we try to ensure the link text has context for the destination. :)

File: apps/labs/posts/doctesting-for-pydata-libraries.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+## Future Goals
+I am currently in the process of integrating the plugin into SciPy; for more details, you can check out the PR linked [here](https://github.com/scipy/scipy/pull/19242). Looking ahead, our goal is to publish the plugin on [PyPI](https://pypi.org/) and extend its integration to NumPy and other PyData libraries. 
+
+If you run into challenges with floating-point arithmetic, face output issues related to whitespace and array abrreviations, need to validate example source code without output testing, or simply desire a customized doctesting experience, consider giving this [plugin](https://github.com/ev-br/scpdt) a try.
+
+## The Journey's End
+Throughout this incredible journey, I cherished every moment spent working, learning from my mentors: [Evgeni Burovski](https://github.com/ev-br) and [Melissa Weber Mendonça](https://github.com/melissawm), and being part of the Quansight team. I'm incredibly grateful for this opportunity, and I look forward to continuing my contributions to the pytest plugin even after the internship.
+
+Curious? Check out the plugin [here](https://github.com/ev-br/scpdt). Feel free to contribute – the more, the merrier! 🚀🐍
```
Comment (pavithraes) [inline apps/labs/posts/doctesting-for-pydata-libraries.md:88]:
```suggestion
Curious? Check out the [plugin repository on GitHub](https://github.com/ev-br/scpdt). Feel free to contribute – the more, the merrier! 🚀🐍
```

**Review by pavithraes — COMMENTED** [review]
Thanks for the post, @Sheila-nk! It looks good to me. Besides a couple of metadata issues, all suggestions are minor. :)

**Comment by Sheila-nk** [comment]
Thank you for the review @pavithraes 
I have included all suggestions in one commit.

**Review by melissawm — APPROVED** [review]
Love this @Sheila-nk! This looks good to go for me :tada:

**Comment by melissawm** [comment]
@ev-br any final comments, or can we merge? thanks!

**Comment by ev-br** [comment]
Looks good to me!

**Comment by melissawm** [comment]
Awesome - I will then merge and this will be published according to the website review schedule. Thanks again @ev-br and @Sheila-nk !

**Pull request merged**